ui/src/conceptList.tsx

```diff
@@ -0,0 +1,103 @@
+import Typography from "@mui/material/Typography";
+import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import React, { useContext, useEffect, useMemo, useState } from "react";
+import { useParams } from "react-router-dom";
+import { EntityInstancesApiContext } from "./apiContext";
+import { State } from "./state";
+
+// Row acts as a bridge between the data format returned from the API and the
+// format expected by DataGrid.
+class Row {
+  id = 0;
+  [x: string]: string | number | boolean | undefined;
+}
+
+type ConceptListProps = {
+  underlay: string;
+  state: State;
+  filter: string;
+};
+
+export default function ConceptList(props: ConceptListProps) {
+  const [error, setError] = useState<Error | null>(null);
+  const [rows, setRows] = useState<Array<Row> | null>(null);
+
+  const params = useParams<{ group: string; criteria: string }>();
+  const api = useContext(EntityInstancesApiContext);
+
+  const columns: GridColDef[] = useMemo(() => {
+    return [
+      { field: "concept_name", headerName: "Concept Name", width: 400 },
+      { field: "concept_id", headerName: "Concept ID", width: 100 },
+      { field: "domain_id", headerName: "Domain", width: 100 },
+    ];
+  }, []);
+
+  useEffect(() => {
+    api
```

Comment:
A pattern you will likely want to establish early is for handling of async cleanup on dismount (avoids this XHR completing after you've already navigated away, and trying to update some stale state or causing side-effects/errors). In Workbench, we use `AbortController` directly ([example](https://github.com/all-of-us/workbench/blob/master/ui/src/app/pages/analysis/runtime-panel.tsx#L1209-L1219)). In Hopper, they use `useAsync`, which I believe handles this as well.